tests/functional/test_plugins_config.py

```diff
@@ -10,299 +10,233 @@
-            assert_cluster_config(cluster_bundle, expected_state)
-
-
-def test_service_config(cluster_bundle: Bundle, keys_clusters_services):
-    expected_state = copy.deepcopy(INITIAL_CLUSTERS_CONFIG)
-    assert_cluster_config(cluster_bundle, expected_state)
-    with allure.step('Check service keys'):
-        for key, cname, sname in keys_clusters_services:
-            cluster = cluster_bundle.cluster(name=cname)
-            service = cluster.service(name=sname)
-            service.action(name='service_' + key).run().try_wait()
-            expected_state[cname]["services"][sname][key] = NEW_VALUES[key]
-            assert_cluster_config(cluster_bundle, expected_state)
-
-
-INITIAL_PROVIDERS_CONFIG = {
-    'first': {
-        'config': copy.deepcopy(INITIAL_CONFIG),
-        'hosts': {
-            'first_First': copy.deepcopy(INITIAL_CONFIG),
-            'first_Second': copy.deepcopy(INITIAL_CONFIG),
-        },
-    },
-    'second': {
-        'config': copy.deepcopy(INITIAL_CONFIG),
-        'hosts': {
-            'second_First': copy.deepcopy(INITIAL_CONFIG),
-            'second_Second': copy.deepcopy(INITIAL_CONFIG),
-        },
-    },
-    'third': {
-        'config': copy.deepcopy(INITIAL_CONFIG),
-        'hosts': {
-            'third_First': copy.deepcopy(INITIAL_CONFIG),
-            'third_Second': copy.deepcopy(INITIAL_CONFIG),
-        },
-    },
-}
-
-PROVIDERS = list(INITIAL_PROVIDERS_CONFIG.keys())
-
-
-def sparse_matrix(*vectors):
-    lengths = []
-    for vector in vectors:
-        lengths.append(len(vector))
-
-    max_lengths_vector_idx = lengths.index(max(lengths))
-    for i, _ in enumerate(vectors[max_lengths_vector_idx]):
-        tmp = []
-        for j, vector in enumerate(vectors):
-            tmp.append(vector[i % lengths[j]])
-        yield tuple(tmp)
-
-
-@allure.step('Check provider config')
-def assert_provider_config(bundle: Bundle, statemap: dict):
-    for pname, plv in statemap.items():
-        actual_cnf = bundle.provider(name=pname).config()
-        expected_cnf = plv['config']
-        for k, v in expected_cnf.items():
-            expect(
-                v == actual_cnf[k],
-                'Provider {} config "{}" is "{}" while expected "{}"'.format(pname, k, str(actual_cnf[k]), str(v)),
+def two_providers(sdk_client_fs: ADCMClient) -> Tuple[Provider, Provider]:
+    """Get two providers with two hosts"""
+    return create_two_providers(sdk_client_fs, __file__, "provider")
+
+
+@pytest.mark.parametrize(
+    ('change_action_name', 'object_to_be_changed', 'action_owner'),
+    generate_cluster_success_params(action_prefix='change', id_template='change_{}_config'),
+)
+@pytest.mark.usefixtures("two_clusters")
+def test_cluster_related_objects(
+    change_action_name: str,
+    object_to_be_changed: Tuple[str, ...],
+    action_owner: Tuple[str, ...],
+    sdk_client_fs: ADCMClient,
+):
+    """
+    Check that correct multi_state_set calls can run successfully
```

Comment:
multistate artifacts

Reply:
Searched for another and fixed what found